includes/collection/class-extra-fields.php

```diff
@@ -231,6 +238,64 @@ function( $rel ) {
 		return $extra_fields;
 	}
 
+	public static function get_extra_fields_for_mastodon_api( $user_id ) {
+		$ret    = array();
+		$fields = self::get_actor_fields( $user_id );
+
+		foreach ( $fields as $field ) {
+			$ret[] = array(
+				'name'   => $field->post_title,
+				'value'  => self::get_formatted_content( $field ),
+			);
+		}
+
+		return $ret;
+	}
+
+	private static function make_paragraph_block( $content ) {
+		if ( ! site_supports_blocks() ) {
+			return $content;
+		}
+		return '<!-- wp:paragraph --><p>' . $content . '</p><!-- /wp:paragraph -->';
+	}
+
+	public static function set_extra_fields_from_mastodon_api( $user_id, $fields ) {
```

Comment:
I think we should have all "enable mastodon apps" functions in the integration file and not mix them up with the "core" files!?!

Reply:
Yup sounds good to me